aea/helpers/dialogue/base.py

```diff
@@ -34,6 +35,9 @@
 from aea.protocols.base import Message
 
 
+NONCE_BYTES_NB = 32
```

Comment:
Why 32?

- Here it is considered sufficient to use 256 bits -> 32 bytes: https://docs.python.org/3/library/secrets.html#how-many-bytes-should-tokens-use
- too many bytes would make it too expensive in terms of network resources (although unless #1643 is fixed, the message field for the nonce will actually take at least twice of the bytes, because the nonce is in hexadecimal format.)
- According to the birthday bound, to have a probability of collision of 50% with 256 bits it is required to have ~ `2^(n/2)`, i.e. `2^128`, which is a massive number. 
Put in other terms: the following is the needed number of messages to have a collision probability of `1e-9` with `256` bits:

```python
def n_approx(H, p):
    return math.sqrt(2 * H * math.log(1 / (1 - p)))

n_approx(2**256, 1e-9)
# 1.5217890705357334e+34
```

Reply:
Another good argument for `32` bytes and not more is the following, stated here: https://man7.org/linux/man-pages/man2/getrandom.2.html

> If the urandom source has been initialized, reads of up to 256 bytes will always return as many bytes as requested and will not be interrupted by signals.  No such guarantees apply for larger buffer sizes.  For example, if the call is interrupted by a signal handler, it may return a partially filled buffer, or fail with the error EINTR.